static/js/filter.js

```diff
@@ -1,430 +1,176 @@
-        
-        filterInput.addEventListener("keydown", e => {
-          if(e.keyIdentifier=="U+000A"||e.keyIdentifier=="Enter"||e.keyCode==13){
-            e.preventDefault();
-            return false;
-          }
-        });
-
-        // TODO: Add esc key listener
-        filterResetButton.addEventListener("click", ()=> {
-            resetFilter();
-        }, false);
-	}
-
-    function resetFilter() {
-        filterInput.classList.remove("is-filtered");
-        filterInput.value = "";
-
-        filterInput.addEventListener("focus", buildSearchQueryArrays, false);
-
-        filterContainer.classList.remove("is-filtered-by-search");
-
-        const isCategoryFilterActive = (filterContainer.classList.contains("is-filtered-by-category"));
-
-        let availableAliasesForSearchFilter = aliases;
-
-        if (isCategoryFilterActive && currentFilteredByCategoryAliases && currentFilteredByCategoryAliases.length > 0) {
-            availableAliasesForSearchFilter = currentFilteredByCategoryAliases;
+    };
+    const categoryMenuEscListener = (event) => {
+        if (event.key === "Escape") {
+            closeCategoryMenu();
         }
-
-        filterLabelVisibleCases.textContent = availableAliasesForSearchFilter.length;
-        filterLabelTotalCases.textContent = availableAliasesForSearchFilter.length;
-
-        availableAliasesForSearchFilter.forEach(alias => {
-            alias.classList.remove("is-hidden");
-        });
+    };
+    function openCategoryMenu() {
+        filterToggleCategoryInput.classList.add("is-enabled");
+        filterCategoryWrapper.classList.add("is-menu-open");
+        filterCategoryCheckboxes[0].focus();
+        document.addEventListener("click", categoryMenuOpenListener, false);
+        document.addEventListener("keydown", categoryMenuEscListener, false);
     }
-
-    // TODO: Remove timeout and watch for event to detect if add-on is enabled (checking if labels exist)
-    setTimeout(filterInit, 500);
-
-    const filterCategoryCheckboxes = document.querySelectorAll(".js-filter-category-checkbox");
-    const filterToggleCategoryButtonReset = document.querySelector(".js-filter-category-reset");
-    const filterToggleCategoryButtonApply = document.querySelector(".js-filter-category-apply");
-
-    function toggleAliasCategoryBar() {
-        filterToggleCategoryInput.classList.toggle("is-enabled");
-        filterCategoryWrapper.classList.toggle("is-menu-open");
-
+    function closeCategoryMenu() {
+        filterToggleCategoryInput.classList.remove("is-enabled");
+        filterCategoryWrapper.classList.remove("is-menu-open");
+        document.removeEventListener("click", categoryMenuOpenListener, false);
+        document.removeEventListener("keydown", categoryMenuEscListener, false);
+    }
+    // TODO: Add "f" key listener to toggle category filter
+    filterToggleCategoryInput.addEventListener("click", (event) => {
```

Comment:
nit: unused event arg
```suggestion
    filterToggleCategoryInput.addEventListener("click", () => {
```

Reply:
Fixed in 19bf72a48b16d8c1408f2229478cb62040bbe1dc.